src/algebraic_geometry/presheafed_space.lean

```diff
@@ -0,0 +1,173 @@
+-- Copyright (c) 2019 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison
+import category_theory.instances.Top.presheaf
+
+universes v u
+
+open category_theory
+open category_theory.instances
+open category_theory.instances.Top
+open topological_space
+
+variables (C : Type u) [𝒞 : category.{v+1} C]
+include 𝒞
+
+namespace algebraic_geometry
+
+structure PresheafedSpace :=
+(X : Top.{v})
+(𝒪 : X.presheaf C)
+
+variables {C}
+
+namespace PresheafedSpace
+
+instance : has_coe_to_sort (PresheafedSpace.{v} C) :=
+{ S := Type v, coe := λ F, F.X.α }
+
+instance (F : PresheafedSpace.{v} C) : topological_space F := F.X.str
+
+structure hom (F G : PresheafedSpace.{v} C) :=
+(f : F.X ⟶ G.X)
+(c : G.𝒪 ⟶ f _* F.𝒪)
+
+@[extensionality] lemma ext {F G : PresheafedSpace.{v} C} (α β : hom F G)
+  (w : α.f = β.f) (h : α.c ≫ (whisker_right (nat_trans.op (opens.map_iso _ _ w).inv) F.𝒪) = β.c) :
+  α = β :=
+begin
+  cases α, cases β,
+  dsimp at w,
+  dsimp [presheaf.pushforward] at *,
+  tidy, -- TODO including `injections` would make tidy work earlier.
+end
+.
+
+def id (F : PresheafedSpace.{v} C) : hom F F :=
+{ f := 𝟙 F.X,
+  c := ((functor.left_unitor _).inv) ≫ (whisker_right (nat_trans.op (opens.map_id _).hom) _) }
+
+def comp (F G H : PresheafedSpace.{v} C) (α : hom F G) (β : hom G H) : hom F H :=
+{ f := α.f ≫ β.f,
+  c := β.c ≫ (whisker_left (opens.map β.f).op α.c) }
+
+variables (C)
+
+section
+local attribute [simp] id comp presheaf.pushforward
+
+instance category_of_PresheafedSpaces : category (PresheafedSpace.{v} C) :=
+{ hom  := hom,
+  id   := id,
+  comp := comp,
+  -- I'm still grumpy about these proofs.
+  -- When I turned the category of open sets upside down by hand,
+  -- I could just leave these out.
```

Comment:
I guess it would help a bit if we could teach `tidy` when to use `op_induction`. But that might be hard... haven't thought about it yet.

Reply:
Until we understand why the `erw`s here are necessary, and make them unnecessary, `tidy` can't do these proofs. If we can get rid of the `erw`s, then it's a simple matter of `local attribute [tidy] op_induction`.